File: src/Knight.java
import java.util.ArrayList;
import java.util.List;

public class Knight extends Piece{
    public Knight(Color color) {
        super(color, 3,"♘","♞");
    }

    @Override
    public List<List<Event>> getValidMoves(Board board, Position position) {
        return new ArrayList<>();
    }
}

File: src/pieces/Knight.kt
package pieces

import Board
import Color
import Piece
import Position
import Tempo
import kotlin.math.abs

class Knight(color: Color) : Piece(color, 3, "♞") {
    override fun getValidMoves(board: Board, position: Position, checkValidation: Boolean): List<Tempo> {
        return board.getAllKnightMoves(position, checkValidation);
    }

    override fun couldBeCheck(board: Board, from: Position, to: Position): Boolean {
        val distance = from.distanceFrom(to)
        return 2 < distance && distance < 2.5
    }

    override fun isCheckFast(board: Board, from: Position, to: Position): Boolean {
        val rowOffset = abs(from.row - to.row)
        val colOffset = abs(from.col - to.col)
        return ((rowOffset == 2 && colOffset == 1) || (rowOffset == 1 && colOffset == 2))
    }
}